src/Symfony/Component/Security/Guard/GuardAuthenticatorHandler.php

```diff
@@ -121,7 +121,7 @@ public function setSessionAuthenticationStrategy(SessionAuthenticationStrategyIn
 
     private function migrateSession(Request $request, TokenInterface $token, ?string $providerKey)
     {
-        if (!$this->sessionStrategy || !$request->hasSession() || !$request->hasPreviousSession() || \in_array($providerKey, $this->statelessProviderKeys, true)) {
+        if (\in_array($providerKey, $this->statelessProviderKeys, true) || !$this->sessionStrategy || !$request->hasSession() || !$request->hasPreviousSession()) {
```

Comment:
Which call starts the session in your case? to me, none of them should.
This means I'm not sure that's the right patch for your issue (but I may very likely miss something).

Reply:
`Request::hasPreviousSession` calls `Request::getSession`, which in turn executes the session factory. So this only happens when the request has a session factory. I tried to find a way to prevent the session factory from being set, but that did not appear to be done so easily.

Reply:
ok got it, thanks for the hints.

Can you please rebase your branch on top of 3.4?
That will fix also the issue reported by fabbot, which is fixed upstream.

Reply:
I may be understanding you wrong, but I don't think the issue exists in the 3.4 branch, since the session is already instantiated when injected into request.

Reply:
symfony/security-guard v3.4 is compatible with symfony/http-foundation v4+, so that this issue can also happen "on 3.4"

Reply:
Ah yes, my bad. I'll have to figure out how to rewrite the test then

Reply:
I see that you've already done my work for me, thanks